### PR TITLE
add option to disable browser cache for security reasons.

### PR DESCRIPTION
**Pull request by tessi**
fixes: https://www.openproject.org/issues/1748

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -104,6 +104,19 @@ def default_url_options(options={})
     { :layout => params["layout"] }
   end
 
+  # set http headers so that the browser does not store any
+  # data (caches) of this site
+  # see: https://websecuritytool.codeplex.com/wikipage?title=Checks#http-cache-control-header-no-store
+  # see: http://stackoverflow.com/questions/711418/how-to-prevent-browser-page-caching-in-rails
+  def set_cache_buster
+    if OpenProject::Configuration['disable_browser_cache']
+      binding.pry
```
Comment (cratz) [inline app/controllers/application_controller.rb]: This line should not be necessary right? ;)

Comment (tessi) [inline app/controllers/application_controller.rb]: oops :]

**Comment by mfrister** [comment]
Looks pretty good and works :)

What do you think about [enabling this by default](https://en.wikipedia.org/wiki/Secure_by_default)? If we don't do this, most people will not enable it, but I think it's a good idea for most people to do.  GitHub e.g. disables caching for all pages, even Google does this for their search homepage on SSL (no idea about the non-SSL version).

We already have `Cache-Control: private, max-age=0, must-revalidate`, which means the browser already has to do a request to the server and the server has to render the page before being able to decide that is hasn't changed and returning a 304. With most pages' HTML not being crazy large and our server-side rendering not being too fast, I don't think the transfer time really counts in contrast to round-trip plus rendering durations.

The only reason for not disabling caching by default I see is that it might be a problem for high-bandwidth sites with caching proxies for public pages in front. People running such pages on the other hand probably will come across caching headers when having performance problems and find the configuration option. Such a page would probably also need a distinction between logged-in and logged out users, but that's out of scope here.

**Comment by tessi** [comment]
I'm OK with your proposed change. either you change the default, or I'll do on my next office day (friday) :)

**Comment by mfrister** [comment]
Done :)

**Comment by tessi** [comment]
:thumbsup:

**Pull request merged**